ambry-replication/src/main/java/com.github.ambry.replication/ReplicationManager.java

```diff
@@ -625,6 +623,17 @@ private void readFromFileAndPersistIfNecessary(String mountPath) throws Replicat
                   logger.warn("Persisted remote replica host {} and port {} not present in new cluster ", hostname,
                       port);
                 }
+              } else {
+                // If this partition was not found in partitionsToReplicate, it means that the local store corresponding
+                // to this partition could not be started. In such a case, the tokens for its remote replicas should be
+                // reset.
+                tokenWasReset = true;
+              }
+              if (tokenWasReset) {
```

Comment:
Is `shouldPersist` added just to not duplicate incrementing the metrics and logging? I'd prefer not adding another variable - just add a helper method to unify the metrics and logging?